### ISSUE TITLE
[HOLD for payment 2023-09-29] [$1000]  Offline: Deleting edited inlinecode block and codeblock results in extra hyphen (strikethrough)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17365]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Send inline codeblock and edit it. (abc )
2. Send Codeblock and edit it.
3. Go offline
4. Delete the edited codeblocks

## Expected Result: 
Strikethrough should be applied only to text. There should not be any extra hyphens.
## Actual Result: 
Unexpected extra hyphens

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.2.99-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
<img width="955" alt="Untitled" src="https://user-images.githubusercontent.com/43996225/231617650-41a45563-9e92-4ef6-9138-1b43072ff3e7.png">
![Screenshot 2023-04-12 at 14 38 24](https://user-images.githubusercontent.com/43996225/231617665-0b7c021b-2d82-445d-a2ca-220e78bbc69f.png)
![Screenshot 2023-04-12 at 14 38 30](https://user-images.githubusercontent.com/43996225/231617666-1cccd6ac-939b-4e9d-8f0f-1b970af1eae8.png)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @sobitneupane 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681289647224179

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01c51c938ba2ca45e8</li>
        <li>Upwork Job ID: 1648053225730277376</li>
        <li>Last Price Increase: 2023-05-08</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @isabelastisser (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by isabelastisser** [comment]
I don't have the capacity to work on this today, so I will prioritize it next week.

**Comment by isabelastisser** [comment]
I will prioritize this today.

**Comment by isabelastisser** [comment]
Reviewing this now.

**Comment by isabelastisser** [comment]
I was able to reproduce it:

<img width="820" alt="image" src="https://user-images.githubusercontent.com/31258516/232596644-a81fbeab-97c1-447e-8861-631c3d13b375.png">

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01c51c938ba2ca45e8

**Comment by MelvinBot** [comment]
Current assignee @isabelastisser is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @aimane-chnaif (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @AndrewGable (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by ahmedGaber93** [comment]
i think it duplicated #17181

**Comment by AndrewGable** [comment]
Can you confirm and close if so @isabelastisser? Thanks!

**Comment by sobitneupane** [comment]
> i think it duplicated #17181

No, it is not. It's a different issue. The other issue is about the alignment of strikethrough.

The issue here is an extra hyphen present before `(edited)` in inline codeblock and codeblock:
![test](https://user-images.githubusercontent.com/25876548/232658978-a33c8554-bf5a-416e-9ea8-f65c6ae8def0.png)

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
There is an extra strikethrough on a pending delete text. It's not only happen to code block, but to all text.

### What is the root cause of that problem?
The space between (edited) and the text is a whitespace character, so it also get the strikethrough style.
https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L18-L27
https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/pages/home/report/ReportActionItemFragment.js#L133-L138

### What changes do you think we should make in order to solve the problem?

Instead of using space character, we can replace it with an empty `View` with the same style, which is styles.w1 (width: 4).

**Comment by dukenv0307** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Unexpected extra hyphens in the `edited` text when deleting edited message in offline mode.

### What is the root cause of that problem?
In here https://github.com/Expensify/App/blob/4019be1c2d41bb27bab90544a84dff7c502f1d3c/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L25 and here https://github.com/Expensify/App/blob/4019be1c2d41bb27bab90544a84dff7c502f1d3c/src/pages/home/report/ReportActionItemFragment.js#L137 we're adding a space at the beginning of the `edited` label to add some spacing with the text message. 

The empty space will get the strike through CSS style as well show it shows as such.

### What changes do you think we should make in order to solve the problem?

We need to apply `styles.textDecorationNoLine` to the empty space `Text` before the `edited` text in the location mentioned above. (in [here](https://github.com/Expensify/App/blob/4019be1c2d41bb27bab90544a84dff7c502f1d3c/src/pages/home/report/ReportActionItemFragment.js#L137) the custom Text component for the empty space isn't there yet so will need to be added). 

This will ensure that we maintain the existing style and also maintain the same behavior when we try to copy both the message and the `edited` text, same as Slack. If we instead use something else other than the empty space for spacing, there'll be an unexpected gap when we try to copy both the message and the `edited` text.

### What alternative solutions did you explore? (Optional)
- We can apply the `styles.textDecorationNoLine` to the whole `edited` text as well if that makes more sense, I don't think there's any use case where we want the strike through style on that `edited` text.

**Comment by isabelastisser** [comment]
Not overdue; still accepting proposals.

**Comment by MelvinBot** [comment]
@AndrewGable, @isabelastisser, @aimane-chnaif Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by isabelastisser** [comment]
@sobitneupane, Should we go ahead with a proposal for this bug?

Also, @sobitneupane @AndrewGable , can you please follow up on @kavimuru question in this[ thread](https://expensify.slack.com/archives/C049HHMV9SM/p1682161364254109) in Slack?

Thanks!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by AndrewGable** [comment]
Let's move forward with issue if it's not a dupe. @aimane-chnaif - Can you review proposals?

**Comment by MelvinBot** [comment]
@AndrewGable @isabelastisser @aimane-chnaif this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by isabelastisser** [comment]
Hey @aimane-chnaif, did you review the proposals? Let's move forward with one, thanks!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by AndrewGable** [comment]
Waiting on @aimane-chnaif to review proposals

**Comment by MelvinBot** [comment]
@AndrewGable @isabelastisser @aimane-chnaif this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by isabelastisser** [comment]
Hi @aimane-chnaif what's next here? Thanks!

**Comment by isabelastisser** [comment]
Hi @aimane-chnaif what's next here? Thanks!

**Comment by AndrewGable** [comment]
Let's see what the update is today and then go from there.

**Comment by isabelastisser** [comment]
Hi @aimane-chnaif , please provide an update. Thanks!

**Comment by isabelastisser** [comment]
Bump @aimane-chnaif

**Comment by isabelastisser** [comment]
any updates @aimane-chnaif ? Thanks!

**Comment by isabelastisser** [comment]
@AndrewGable can you please follow up on @aimane-chnaif comment here:

> https://github.com/Expensify/App/issues/17365#issuecomment-1567212517

**Comment by AndrewGable** [comment]
@aimane-chnaif - Thoughts on https://github.com/Expensify/App/issues/17365#issuecomment-1512830755 ?

**Comment by koko57** [comment]
Hi 🙂 I've done some investigation on this issue and here are my findings: the [updated proposal](https://github.com/Expensify/App/issues/17365#issuecomment-1512830755) unfortunately won't work here because the strikethrough style is applied higher in the tree

![Screenshot 2023-05-31 at 12 21 08](https://github.com/Expensify/App/assets/28020445/94373706-4205-45a5-babd-cd8dc771847c)

(I've enlarged the edited label on purpose)

I've tried @bernhardoj [solution](https://github.com/Expensify/App/issues/17365#issuecomment-1512448803) - I've added the View component above the Text, removed the element with the space and wrapped it in a fragment. It didn't work at first - there was no space between the codeblock and the 'edited' label.

![Screenshot 2023-05-31 at 12 34 20](https://github.com/Expensify/App/assets/28020445/f172b598-17fc-4f5f-bc29-bef1b69c122b)

But I've found the culprit here - in this PR the styles' w1 property was completely removed (I've also checked these changes applied here but they also had the problem with the space with the strikethrough). Then the changes were partially reverted [here](https://github.com/Expensify/App/commit/16d5cf17d44760e6516963096d4eb380fde8180c), but the style property wasn't restored. 

So restoring the w1 property in [sizing](https://github.com/Expensify/App/blob/efb1463625ba784999dce58dec0faa9163e33e0b/src/styles/utilities/sizing.js) styles makes the @bernhardoj solution with the View work. It looks like this -  we have an additional div here with a proper width - no space in text => no striketrough

![Screenshot 2023-05-31 at 12 36 07](https://github.com/Expensify/App/assets/28020445/507d61c7-0584-409d-bc8f-d80d5153b0bc)
![Screenshot 2023-05-31 at 12 48 27](https://github.com/Expensify/App/assets/28020445/cff8abed-65e3-4488-bd12-2ffea6f8d308)

I've also found another working solution for this one: adding `paddingLeft: 4`. We don't need to add an additional element here. 

![Screenshot 2023-05-31 at 13 02 35](https://github.com/Expensify/App/assets/28020445/7635d19a-b695-4e40-a43c-34c5c59be165)

![Screenshot 2023-05-31 at 12 57 13](https://github.com/Expensify/App/assets/28020445/61a914df-257e-46ca-9290-e137ab3b81d5)

![Screenshot 2023-05-31 at 13 00 58](https://github.com/Expensify/App/assets/28020445/76efa60a-4963-40a2-868f-87635cf56daf)

**Comment by koko57** [comment]
@aimane-chnaif Yeah, sorry you're right - the padding solution doesn't work here (no padding).

![Screenshot 2023-05-31 at 13 41 22](https://github.com/Expensify/App/assets/28020445/dbeb1895-df2c-4145-9baf-976faf26d2ae)

 But the View solution seems to be working

![Screenshot 2023-05-31 at 13 44 07](https://github.com/Expensify/App/assets/28020445/e1e431d6-2468-4791-bd3c-06e0738e27cf)

**Comment by bernhardoj** [comment]
Is the usage of View does not work because of the View inside Text issue we have? If yes, then we should just wait for https://github.com/facebook/react-native/pull/35703.

**Comment by AndrewGable** [comment]
@aimane-chnaif please review above, thank you!

**Comment by koko57** [comment]
@dukenv0307 yeah, I've just ckecked your solution works now for web (I wonder why, as it didn't work when I was tested it previously). But unfortunately it still doesn't work on iOS. @dukenv0307 did you test it for other platforms?
![Simulator Screen Shot - iPhone 14 - 2023-06-05 at 10 43 50](https://github.com/Expensify/App/assets/28020445/51408aa6-fed8-4143-8c75-739b47a43564)

@aimane-chnaif ok, i'll see if I can improve this solution or if it still won't be working, I'll look for a new one 🙂

**Comment by koko57** [comment]
@aimane-chnaif I've checked the View solution once again - when properly applied (both in EditedRenderer and ReportActionItemFragment) it works for all the platforms. But I agree it's not an ideal solution, but still, it works.

Android Chrome
![Screenshot_1686155003](https://github.com/Expensify/App/assets/28020445/37d94576-1260-43c6-ae62-f68a20602603)

Android Native
![Screenshot_1686155324](https://github.com/Expensify/App/assets/28020445/6a32ce9e-7a7a-41f8-9af3-dd3feea28e06)

iOS Safari
![Simulator Screen Shot - iPhone 14 - 2023-06-07 at 18 29 33](https://github.com/Expensify/App/assets/28020445/ab5cd35d-907b-418a-bed3-261bb3e58b75)

iOS Native
![Simulator Screen Shot - iPhone 14 - 2023-06-07 at 18 29 09](https://github.com/Expensify/App/assets/28020445/860b4f02-0bf1-4880-8401-ff8ce1c832db)

Desktop
![Screenshot 2023-06-07 at 18 21 08](https://github.com/Expensify/App/assets/28020445/0c5e792e-4867-4241-8c6c-5c960382f3bc)

Web Brave
![Screenshot 2023-06-07 at 18 22 26](https://github.com/Expensify/App/assets/28020445/cf5157f4-8b85-49f8-aca3-fec474c710b4)

Web Safari
![Screenshot 2023-06-07 at 18 22 00](https://github.com/Expensify/App/assets/28020445/0577b14b-d13c-4bef-a329-c09a4a157aa3)

**Comment by koko57** [comment]
Today I did some more investigation on how the deleted styles are applied. The strikethrough styles are applied to the ReportActionItemSingle and ReportActionItemGrouped - it is done by the [applyStrikeThrough](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/OfflineWithFeedback.js#L72) in [OfflineWithFeedback](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/OfflineWithFeedback.js#L85) (so way higher in the hierarchy than the text components we want the styles applied to). I thought that maybe we could move this logic down to the components that are responsible for rendering the text that should get the strikethrough styles. But as OfflineWithFeedback is used in many location, that would require a lot of work and I'm not 100% sure, if it would work for all the components wrapped in OfflineWithFeedback. 

I came up with a solution to move the logic only for the ReportActionItems, as we already kind of doubling it  in [applyStrikethough](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/HTMLEngineProvider/applyStrikethrough/index.native.js#L1) for native and the styles in [EditedRenderer](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/HTMLEngineProvider/HTMLRenderers/EditedRenderer.js#L25). 

What I've done:

- moved the styles to the 'edited' tag like in situchan [solution](https://github.com/Expensify/App/issues/17365#issuecomment-1512830755) in both EditedRenderer and [ReportActionItemFragment](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/pages/home/report/ReportActionItemFragment.js#L127-L141)
- added a prop 'noStrikeThrough' to ReportActionItemGrouped and ReportActionItemSingle 
- added a check for this prop to this [condition](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/OfflineWithFeedback.js#LL74C13-L74C13) in applyStrikeThrough function to return unchanged child.
- used the logic from native [applyStrikethough](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/HTMLEngineProvider/applyStrikethrough/index.native.js#L1) for adding a <del></del> tag to [applyStrikethough](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/HTMLEngineProvider/applyStrikethrough/index.js#L1) used for web
- applied my solution from the previous comment `const htmlContent = applyStrikethrough(html, true) + editedTag;
`

Yeah, this solution seems to be way too complicated and looks like could cause a lot of regression, but maybe it could be the first step to refactor adding the strikethrough styles for all the components. Still checking for any regression, but at the moment it seems to be working

Android Native
![Screenshot_1686153453](https://github.com/Expensify/App/assets/28020445/3c6d6d92-e347-47ed-bab7-7d6003518645)

Android Chrome
![Screenshot_1686153476](https://github.com/Expensify/App/assets/28020445/cf3f07a7-362e-474b-8b19-a42d4802e967)

iOS Safari
![Simulator Screen Shot - iPhone 14 - 2023-06-07 at 17 58 04](https://github.com/Expensify/App/assets/28020445/6d43ea55-d78e-4801-a684-7f6a2c19f07b)

iOS Native
![Simulator Screen Shot - iPhone 14 - 2023-06-07 at 17 58 20](https://github.com/Expensify/App/assets/28020445/3144c2f2-e83b-49ad-9c16-d6f0b761b302)

Desktop
![Screenshot 2023-06-07 at 17 56 32](https://github.com/Expensify/App/assets/28020445/1c26d90b-9517-4f31-a89d-56ed87dd01c8)

Web Safari
![Screenshot 2023-06-07 at 17 56 55](https://github.com/Expensify/App/assets/28020445/8f804b13-754b-498b-866f-673a039d8124)

Web Brave
![Screenshot 2023-06-07 at 17 57 23](https://github.com/Expensify/App/assets/28020445/340579a5-5312-4db3-b4a8-88b2ff4a6efd)

That's all that I came up with during my investigations.

I'll be OOO until the end of this week, I will get back to it on Monday.

**Comment by AndrewGable** [comment]
Waiting for the perfect proposal still!

**Comment by koko57** [comment]
Still looking for a better solution

**Comment by koko57** [comment]
Today I was working on another issue, I should get back to this one tomorrow

**Comment by isabelastisser** [comment]
Internal, not overdue!

**Comment by koko57** [comment]
I still haven't found a better solution. I've checked once again - @bernhardoj solution with view still works for all the platforms and is easy to implement. Diff here https://github.com/Expensify/App/compare/main...koko57:App:fix/17365-strikethrough-view-solution?expand=1 

But still I think that we shouldn't apply these strikethrough styles with the [applyStrikeThrough](https://github.com/Expensify/App/blob/bd5db762cd07a0258d92462d8ddaffc02315854e/src/components/OfflineWithFeedback.js#L72) function in the OfflineWithFeedback as it adds those styles unnecessarily to the parent components that are ascendants for the component, that we actually want to give the styles. So the View solution is good for that particular case, it solves the problem, but it doesn't solve the root cause of this issue. In the future, if we wanted to wrap a new component with the OfflineWithFeedback there may be some other problems with this approach. 

I'm still thinking how to move this logic to the exact component we want to strikethrough without prop drilling  or causing regression for other component wrapped in OfflineWithFeedback.

**Comment by koko57** [comment]
Still looking for a better solution

**Comment by isabelastisser** [comment]
not overdue, internal!

**Comment by koko57** [comment]
Testing a solution quite similar to the one I've proposed earlier, I'll let you know if it works

**Comment by koko57** [comment]
still working on it

**Comment by isabelastisser** [comment]
No updates!

**Comment by koko57** [comment]
I'm really close to propose another solution (quite similat but a bit simplier than one I've proposed earlier) but I've got one problem. 

@bernhardoj good you're here in the conversation 😃 - in your [PR](https://github.com/Expensify/App/pull/20114/files) - do we really need to add `styles.alignItemsStart` to [defaultViewProps](https://github.com/Expensify/App/blob/6183609c577f2310928bb4e8ee23b52f6dd28e00/src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js#L51) here? Could it be done like it was before the changes (adding to the components that really need it)? Do we have to add it to all the views?

**Comment by isabelastisser** [comment]
Assigning another team member because I will be OOO until July 5. [SO ](https://stackoverflowteams.com/c/expensify/questions/8836)for reference.

**Comment by maddylewis** [comment]
got it! i can keep an eye on this one, @isabelastisser! in the future, please try to include a detailed description of what's needed from the BZ member assigned. that makes things easier from this perspective. [here's](https://github.com/Expensify/App/issues/17776#issuecomment-1590232940) an example for you to reference!

**Comment by maddylewis** [comment]
@aimane-chnaif / @bernhardoj - when you have a chance, can you take a look at @koko57's question here - https://github.com/Expensify/App/issues/17365#issuecomment-1604439120

**Comment by koko57** [comment]
@maddylewis @aimane-chnaif @bernhardoj It's ok, today I've found a solution that won't affect any changes made in this PR I asked about 🙂 
I will post an update at the EOD

**Comment by bernhardoj** [comment]
Oh, my bad. I totally missed this.

For context, previously the `styles.alignItemsStart` is there, but then get removed in this PR https://github.com/Expensify/App/issues/16526. 

> (adding to the components that really need it)
I initially want to do it like this to solve this issue https://github.com/Expensify/App/issues/19717, but then we have this issue https://github.com/Expensify/App/issues/19934 that changes my mind. I explain the detail in my proposal here why we need to have the `styles.alignItemsStart` back https://github.com/Expensify/App/issues/19934#issuecomment-1571918043.

Sorry for the late response 🙇

**Comment by koko57** [comment]
@bernhardoj It's ok 🙂 thanks for the explanation. 

So my update here: it was really really hard for me to find a way to gracefully refactor this logic for all the components wrapped in the OfflineWithFeedback. I simplified my earlier [solution](https://github.com/Expensify/App/compare/main...koko57:App:fix/17365-strikethrough-refactor-solution?expand=1) and make it work with the changes introduced in this [PR](https://github.com/Expensify/App/pull/20114/files), because after merging it didn't work.

What's happening [here](https://github.com/Expensify/App/compare/main...koko57:App:fix/17365-nostrikethrough-solution?expand=1):
- I applied the noStrikeThrough prop to [OfflineWithFeedback](https://github.com/koko57/App/blob/30adf9a10cd44c7125897ddb4035c43b939893a4/src/pages/home/report/ReportActionItem.js#L476-L484) - only for the ReportActionItem (as we want to deal with it differently)
- checking for the prop [here](https://github.com/koko57/App/blob/30adf9a10cd44c7125897ddb4035c43b939893a4/src/components/OfflineWithFeedback.js#L100) to ensure if we need the strikethrough applied in the OfflineWithFeedback
- as we're already checking if isPendingDelete in the [ReportActionItemFragment](https://github.com/koko57/App/blob/30adf9a10cd44c7125897ddb4035c43b939893a4/src/pages/home/report/ReportActionItemFragment.js#L97) I just moved it to make it available for almost the whole 'COMMENT' case.
- I deleted the applyStrikethrough function that was different for the native and for the web and applied the same logic in a simplier way [here](https://github.com/koko57/App/blob/30adf9a10cd44c7125897ddb4035c43b939893a4/src/pages/home/report/ReportActionItemFragment.js#L115) - it works both for the native and the web
- moved the 'deleted' styles to the 'edited' tag text in EditedRenderer and [here](https://github.com/koko57/App/blob/30adf9a10cd44c7125897ddb4035c43b939893a4/src/pages/home/report/ReportActionItemFragment.js#L124-L149), made the Text with the space a sibling to the Text with 'edited' tag
- added one style improvement (because without that line the UI is broken for the native, because of `alignItems: center` for in the [defaultViewProps](https://github.com/Expensify/App/blob/d1b386979a42613ef9a80ecc9368e10e6c8a81f1/src/components/HTMLEngineProvider/BaseHTMLEngineProvider.js#L77); `flexDirection: 'row'` also works here instead) 
here's why:
![Simulator Screen Shot - iPhone 14 - 2023-06-26 at 17 57 57](https://github.com/Expensify/App/assets/28020445/196af134-466c-4226-a094-266d63e4f7ca)


I didn't found any problems with this solution, it works fine for all the platforms. The logic for not applying strikethrough on the OfflineWithFeedback level could be potentially useful in the future for some new components.

Screenshots:

Desktop:
![Screenshot 2023-06-26 at 18 03 19](https://github.com/Expensify/App/assets/28020445/0641f857-5fb5-487b-9a88-8184e93d15fd)

Web Chrome:
![Screenshot 2023-06-26 at 18 06 23](https://github.com/Expensify/App/assets/28020445/1d74fc19-4298-40ac-b299-c5c4bc4eea64)

Web Safari:
![Screenshot 2023-06-26 at 18 07 19](https://github.com/Expensify/App/assets/28020445/01e5ca0a-8dc3-4987-b6b4-dd62a88518d2)

iOS Native:
![Simulator Screen Shot - iPhone 14 - 2023-06-26 at 18 08 49](https://github.com/Expensify/App/assets/28020445/cff3d809-2e04-4dc2-9d53-7a299e76813d)

iOS Safari:
![Simulator Screen Shot - iPhone 14 - 2023-06-26 at 18 09 40](https://github.com/Expensify/App/assets/28020445/4fdf2a6d-af04-4c2f-9ebc-1b0abf72959a)

Android Native: (here we see two other issues that are not caused by my changes - the inline code not displaying and the cut off emoji)
![Screenshot_1687795969](https://github.com/Expensify/App/assets/28020445/3ed56ee0-e416-48c1-b701-f363b18e3ce2)

Android Chrome:
![Screenshot_1687795842](https://github.com/Expensify/App/assets/28020445/29cde699-b1cc-4adf-8776-fb6dcbde3859)

@aimane-chnaif could you please review it when you have some time and tell me what do you think about this? 🙏🏻

**Comment by maddylewis** [comment]
waiting for @aimane-chnaif's review of this here - https://github.com/Expensify/App/issues/17365#issuecomment-1607809242

**Comment by maddylewis** [comment]
thanks for the update @aimane-chnaif - i will move to `weekly` until that [PR](https://github.com/Expensify/App/pull/18507) is merged.

**Comment by isabelastisser** [comment]
I'm back. Thanks, @maddylewis!

**Comment by isabelastisser** [comment]
@aimane-chnaif any updates?

**Comment by isabelastisser** [comment]
hey @aimane-chnaif any updates?

**Comment by isabelastisser** [comment]
@AndrewGable @aimane-chnaif what are the next steps here?

**Comment by TomaszFrackowiak** [comment]
@aimane-chnaif - hey, just to let you know that @koko57 is ooo, she will be back 25 Aug

**Comment by koko57** [comment]
@aimane-chnaif sorry I forgot to mention in this issue that I'll be off till Friday!If I remember correctly it's because the strikethrough is handled differently on mobile and web/desktop (we have separate component for native)

**Comment by koko57** [comment]
@aimane-chnaif to clarify: as we're applying the strikethrough styles in EditedRenderer we don't want the whole content (html and editedTag) to be additionally struck through by [this](https://github.com/Expensify/App/blob/fd407e256f17df6236f5160c0b240db6d2f00333/src/components/HTMLEngineProvider/applyStrikethrough/index.native.js#L1-L8) We want to apply those styles for the html and editedTag separately

**Comment by koko57** [comment]
@aimane-chnaif Could you please review my latest [solution](https://github.com/Expensify/App/issues/17365#issuecomment-1607809242)?

**Comment by koko57** [comment]
updated proposal: https://github.com/Expensify/App/issues/17365#issuecomment-1578864389

**Comment by isabelastisser** [comment]
Bump @aimane-chnaif to review the latest solution above, thanks!

**Comment by koko57** [comment]
PR: https://github.com/Expensify/App/pull/26176

**Comment by isabelastisser** [comment]
@aimane-chnaif, please complete the checklist above so we can close the issue. Thanks!

**Comment by isabelastisser** [comment]
Thanks @aimane-chnaif ! Closing.

**Comment by sobitneupane** [comment]
Requested payment of $250 as reporter of the bug in newDot.

**Comment by JmillsExpensify** [comment]
Re-opening because I need a payment summary for this issue.

**Comment by JmillsExpensify** [comment]
$250 payment approved for @sobitneupane based on summary above.

**Comment by trjExpensify** [comment]
Cool, thanks! @aimane-chnaif were you paid for this via Upwork as per the checkmarks [here](https://github.com/Expensify/App/issues/17365#issuecomment-1731402443)?

**Comment by trjExpensify** [comment]
Perf, closing!
